### ISSUE TITLE
Shallow clone specific commit with tags

### ISSUE THREAD

**Issue by raballew** [tektoncd/pipeline#3387]
### Feature request

```
resources:
  - name: source-git
    resourceSpec:
      params:
        - name: refspec
          value: "refs/tags/*:refs/tags/*"
        - name: revision
          value: $(params.git-revision)
        - name: url
          value: $(params.git-repository-url)
      type: git
```

If I would use the pipeline resource specification shown above, I would expect that all tags referencing the git revision would be fetched as well. Using plain git commands this could be achieved by the following:

```
git init
git remote add origin <git-repository-url>
git fetch --tags --depth 1 origin <git-revision>
git checkout FETCH_HEAD
git log

commit a90ddc239d543598810322b039049b820a099086 (grafted, HEAD, tag: 0.1.0)
```

But actually the current implementation does not fetch the tags.

```
git log

commit a90ddc239d543598810322b039049b820a099086 (grafted, HEAD)
```

A manual `git fetch --tags` is required in the task using the pipeline resource.

### Use case

A use case is shown in [this hello-world example](https://github.com/raballew/hello-world/tree/b230c598abdbb509d8e58e6d78d1a55b509ceef4).

TLDR; A push event delivers a payload to an event listener. This triggers pipeline `process-push-event`, which iterates over the commits changed by the push event and then triggers another pipeline `process-git-revision`. `process-git-revision` analyses the versions related to a specific git revision in the first task and stores the results as results. As a result one is able to extract version information such as if there is a final, candidate, beta and/or alpha for this git revision. Subsequent tasks then can decide whether they want to deploy to prod, staging or qa with when expressions. This implementation allows a kind of dynamic fan out for push events with a pipeline run for **each commit** with massive parallelism.

**Comment by ad22** [comment]
I added the refspec functionality, so I can help take a look. IIRC this was precisely the use case I needed, and this _should_ work.

**Comment by raballew** [comment]
@ad22 Thanks for helping! Knowing this, I did some more debugging and manually pulled the [latest and greatest image](https://console.cloud.google.com/gcr/images/tekton-releases/GLOBAL/github.com/tektoncd/pipeline/cmd/git-init@sha256:a39b8c647d6eaf609dc04bc7b0fc0f787b3b25fd6b1c9b182066a5c29a64a18c/details?tab=info) and everything looks fine. So I assume that this behaviour is caused by some kind of misconfiguration on my side.

```
$ docker run -it --entrypoint /bin/sh gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/git-init@sha256:a39b8c647d6eaf609dc04bc7b0fc0f787b3b25fd6b1c9b182066a5c29a64a18c
$ export HOME=$pwd
$ /ko-app/git-init -url https://github.com/raballew/hello-world.git -revision a90ddc239d543598810322b039049b820a099086 -refspec "refs/tags/*:refs/tags/*"

{"level":"warn","ts":1602802319.6770737,"caller":"git/git.go:263","msg":"SSH credentials have been provided but the URL(\"https://github.com/raballew/hello-world.git\") is not a valid SSH URL. This warning can be safely ignored if the URL is for a public repo or you are using basic auth"}
{"level":"fatal","ts":1602802319.6817548,"caller":"git-init/main.go:53","msg":"Error fetching git repository: error checking for known_hosts file: open /root/.ssh/known_hosts: too many levels of symbolic links","stacktrace":"main.main\n\tgithub.com/tektoncd/pipeline/cmd/git-init/main.go:53\nruntime.main\n\truntime/proc.go:203"}

$ git log

commit a90ddc239d543598810322b039049b820a099086 (grafted, HEAD, tag: 0.1.0)
```
